round4/TabTracker.md

```diff
@@ -0,0 +1,109 @@
+
+## User Stories
+
+1. As a user, I want to be taken directly to my desired tab by clicking on the tab description in the popup window
+2. As a researcher, I want to have a clear overview of all my open tabs and their content summaries to efficiently navigate between different research topics.
+3. As a user who often multitasks, I want to easily spot tabs that have been open for a long time so that I can prioritize which tabs to focus on.
+4. As a developer, I want to manage tabs related to different coding projects and documentation to enhance my productivity.
+5. As a project manager, I want to easily spot and close tabs related to completed or outdated projects to declutter my browser.
+6. As a user, I can easily manage my tabs to avoid having tabs of the same information.
+7. As a user, I want to be able to set a timer threshold such that TabTracker will close tabs which pass this threshold.
+8. As a student, I want to clearly see which of my tabs are inactive and which ones are active
+9. As a student, I want to easily be able to close multiple inactive tabs at once instead of individually
+10. As a user I want a scroll feature that allows me to easily see all of the tabs I have open
+11. As a teacher with many papers to review online, TabTracker allows me to reduce clutter on my browser by prompting me to close those papers I’ve already graded.
+12. As the CEO of Google, I am tired of Chrome not doing anything to help fix my excessive tab issue and can now have an efficient browsing experience.
+13. As a parent, I can get a prompt to close all the tabs my kid has opened when he is finished using my computer to play video games.
+14. As a student with many classes, I want my tabs in different browsers to show up on a holistic breakdown.
+15. As a student athlete, I can more efficiently get schoolwork done by closing unnecessary outdated sports-related tabs.
+
+## Notes
+
+Certain potential features:
+Ordering tabs by most recently used
+Tab grouping with color-coded columns
+Implementing a search feature
+
+## References & Inspiration
+
+OneTab: <https://chromewebstore.google.com/detail/onetab/chphlpgkkbolifaimnlloiipkdnihall?pli=1>
+
+SessionBuddy -
+<https://chromewebstore.google.com/detail/session-buddy/edacconmaakjimmfgnblocblbcdcpbko>
+
+## Technical Details
+
+### User Interface
+
+Upon activating the extension, a side panel emerges displaying the activity and inactivity of each tab. The names and contents of each tab will be arrayed vertically and ordered by most recently used. The inactive tabs will be marked by red color-coding, whereas the active tabs will be marked by green color-coding. A prompt will also emerge to close all those highlighted red tabs, with an option to deselect some or all tabs to escape deletion. Additionally, hovering over the tabs in the sidebar will allow you to directly access any said tab through clicking on it.
```

Comment:
I think it would also be nice if you add an optional feature where the extension will automatically close tabs that have been inactive for x amount of time. I like how you are already considering design details for how users can manage different tabs, which is an important selling point, since Chrome's tab UI is already pretty good in my opinion. I would also suggest adding filtering by the name of a tab to add more user flexibility. I would also suggest adding an undo button in case users accidentally misclick and delete a tab they wanted to keep.  A lot of groups are doing similar projects, so I'm interested in seeing how each group approaches the problem. I think this is a feasible project, and if you have any extra time, maybe consider some stretch goals? (One idea that comes to mind is to show analytics for the number of concurrent tabs over time or showing the breakdown of what tabs are using and for how long).